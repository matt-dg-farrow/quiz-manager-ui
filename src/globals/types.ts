export type Answer = { id: number; text: string; isCorrect: boolean };
export type Question = { id: number; title: string; answers: Answer[] };
export type Quiz = { id: number; title: string; questions: Question[] };
