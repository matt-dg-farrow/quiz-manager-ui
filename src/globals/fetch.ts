import { Quiz } from "./types";

const API_BASE = "http://localhost:8080/";

const headers = {
  "Content-Type": "application/json; charset=utf-8",
};

export const Post = (url: any, query: any) =>
  fetch(`${API_BASE}${url}`, {
    headers,
    method: "POST",
    credentials: "same-origin",
    body: JSON.stringify(query),
  })
    .then((resp) => resp.json())
    .catch(() => ({
      ok: false,
      data: { statusMessage: "Parsing POST Error" },
    }));

export const Get = (url: any, query?: any) =>
  fetch(`${API_BASE}${url}`, {
    headers,
    method: "GET",
    credentials: "same-origin",
    body: JSON.stringify(query),
  })
    .then((resp) => resp.json())
    .catch(() => ({ ok: false, data: { statusMessage: "Parsing GET Error" } }));

export const Delete = (url: any, query: any) =>
  fetch(`${API_BASE}${url}`, {
    headers,
    method: "DELETE",
    credentials: "same-origin",
    body: JSON.stringify(query),
  })
    .then((resp) => resp.json())
    .catch(() => ({
      ok: false,
      data: { statusMessage: "Parsing DELTE Error" },
    }));

export const Put = (url: any, query: any) =>
  fetch(`${API_BASE}${url}`, {
    headers,
    method: "PUT",
    credentials: "same-origin",
    body: JSON.stringify(query),
  })
    .then((resp) => resp.json())
    .catch(() => ({
      ok: false,
      data: { statusMessage: "Parsing PUT Error" },
    }));
