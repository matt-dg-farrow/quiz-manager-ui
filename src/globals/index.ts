export enum QuizRequestTypes {
  FETCH_QUIZ_DATA_ACTION = "FETCH_QUIZ_DATA_ACTION",
  FETCH_QUIZ_DATA_ACTION_SUCCESS = "FETCH_QUIZ_DATA_ACTION_SUCCESS",
  FETCH_QUIZ_DATA_ACTION_FAILURE = "FETCH_QUIZ_DATA_ACTION_FAILURE",
  FETCH_QUIZ_DATA_REQUEST = "quiz/all",

  CREATE_QUIZ_ACTION = "CREATE_QUIZ_ACTION",
  CREATE_QUIZ_ACTION_SUCCESS = "CREATE_QUIZ_ACTION_SUCCESS",
  CREATE_QUIZ_ACTION_FAILURE = "CREATE_QUIZ_ACTION_FAILURE",
  CREATE_QUIZ_ACTION_REQUEST = "quiz/create",

  DELETE_QUIZ_ACTION = "DELETE_QUIZ_ACTION",
  DELETE_QUIZ_ACTION_SUCCESS = "DELETE_QUIZ_ACTION_SUCCESS",
  DELETE_QUIZ_ACTION_FAILURE = "DELETE_QUIZ_ACTION_FAILURE",
  DELETE_QUIZ_ACTION_REQUEST = "quiz/delete",

  EDIT_QUIZ_ACTION = "EDIT_QUIZ_ACTION",
  EDIT_QUIZ_ACTION_SUCCESS = "EDIT_QUIZ_ACTION_SUCCESS",
  EDIT_QUIZ_ACTION_FAILURE = "EDIT_QUIZ_ACTION_FAILURE",
  EDIT_QUIZ_ACTION_REQUEST = "quiz/update",
}
