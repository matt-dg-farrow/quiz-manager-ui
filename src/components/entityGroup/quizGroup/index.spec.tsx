import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import QuizGroup from ".";
import { Quiz } from "../../../globals/types";

const quizzes: Quiz[] = [
  {
    id: 1,
    title: "Quiz 1",
    questions: [
      {
        id: 2,
        title: "Question 1",
        answers: [
          { id: 3, text: "Answer 1", isCorrect: true },
          { id: 4, text: "Answer 1", isCorrect: true },
          { id: 5, text: "Answer 1", isCorrect: true },
        ],
      },
    ],
  },
];

describe("The QuizGroup component", () => {
  const wrapper = shallow(<QuizGroup quizData={quizzes} />);

  it("should contain the correct number of components", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
