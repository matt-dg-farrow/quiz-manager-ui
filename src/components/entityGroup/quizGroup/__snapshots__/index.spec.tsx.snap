// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The QuizGroup component should contain the correct number of components 1`] = `
<div>
  <div
    key="1"
  >
    <QuizItem
      quiz={
        Object {
          "id": 1,
          "questions": Array [
            Object {
              "answers": Array [
                Object {
                  "id": 3,
                  "isCorrect": true,
                  "text": "Answer 1",
                },
                Object {
                  "id": 4,
                  "isCorrect": true,
                  "text": "Answer 1",
                },
                Object {
                  "id": 5,
                  "isCorrect": true,
                  "text": "Answer 1",
                },
              ],
              "id": 2,
              "title": "Question 1",
            },
          ],
          "title": "Quiz 1",
        }
      }
      view="list"
    />
  </div>
</div>
`;
